doc/development/adding_templates.rst

```diff
@@ -0,0 +1,91 @@
+How to add a new template
+-------------------------
+
+The following steps will help you to add your favourite circuit ansatz to
+PennyLane's :mod:`template <pennylane.template>` library.
+
+PennyLane internally distinguishes :ref:`different types <intro_ref_temp>` of templates, such as
+:mod:`embeddings <pennylane.templates.embeddings>`, :mod:`layers <pennylane.templates.layers>`,
+:mod:`state_preparations <pennylane.templates.state_preparations>` or
+:mod:`soubroutines <pennylane.templates.soubroutines>`.
+Here we will use the template ``MyNewTemplate`` as an example, and you need to replace ``<templ_type>`` with the
+correct template type.
+
+1. **Add the template** by adding a new file ``my_new_template.py`` to the correct ``templates/<templ_type>/``
+   subdirectory, which contains your new template of the form
```

Comment:
Perhaps it might make it might be easier to follow if the guide walked through the process of adding an actual template example. For example,

```python
@template
def MyLayer(weights, wires, entangler=qml.CZ):
    # The template can apply arbitrary quantum operations.
    # The operations applied may depend on the shape of input arguments.

   # iterate through layers
    for layer, p in enumerate(weights):
        for w in wires:
            # apply X rotation to each wire
            qml.RX(w, wires=w)

        # The qml.broadcast function may also be used.
        # Apply entangler to wire pairs that connect all wires to each other.
        qml.broadcast(entangler, wires, pattern="all_to_all")
```

Reply:
The steps below could be made more clear by going back to this example, e.g., "In this example, since we are creating a layer template, we place this function in the file `pennylane/templates/layers/my_layer.py`"

Reply:
That is a nice idea. However, I substantially simplified the example, as this one has a lot of subtleties...

Reply:
Feel free!